pure3.md

```diff
@@ -0,0 +1,282 @@
+strategies that can secure automotive software supply chains are vital to
+_avoid_ such attacks altogether.
+
+Governments around the world have recognized this fact. Regulations are
+forthcoming from the United Nations and other governing bodies, and standards
+have been published that specify how software running on automobiles must be
+secured. President Biden of the United States of America issued an Executive
+Order describing the need to improve the nation's cybersecurity. This EO, in
+particular, pushes for solutions like Software Bills of Materials (SBOMs) that
+list the components of software artifacts. This information is vital in ensuring
+developers are aware of the components in their supply chain so that they can
+fix vulnerabilities as and when they are discovered.
+
+Unfortunately, SBOMs do not provide end-to-end software supply chain security.
+For example, an SBOM would not have prevented an attack that used a compromised
+build server as seen in the SolarWinds hack. A more comprehensive framework is
+necessary and is described in this document.
+
+## Reasoning
+
+in-toto is a comprehensive framework that secures each step in the software
+supply chain. With in-toto, software supply chain owners can enforce policies
+pertaining to the flow of artifacts through the supply chain, and provide proof
+that authorized people or bots made changes to artifacts at any given step.
+Thus, in-toto can be used to secure software from the very first step all the
+way through the development pipeline. Furthermore, the framework plays well with
+other systems and tools that attempt to secure specific aspects of the software
+supply chain. in-toto is integrated into build tools and systems like Jenkins
+and Tekton, and is compatible with complementary software supply chain security
+efforts such as Grafeas, Sigstore, and Software Bills of Materials.
+
+## Specification
+
+Note: The reader is expected to be familiar with the
+[in-toto Specification](https://github.com/in-toto/docs/blob/master/in-toto-spec.md).
+
+We propose in-toto as an option to achieve software supply chain security
+alongside the Uptane deployment. In such a model, in-toto can be backed by
+the Uptane root of trust, and use Uptane's delegation structure to distribute
+all of in-toto's metadata. Essentially, in-toto metadata is used to attest to
+the steps performed in the software supply chain itself, while Uptane is used
+to securely distribute that metadata alongside the ECU images. This combination
+of in-toto for software supply chain guarantees and Uptane for secure
+distribution of ECU images and in-toto metadata is called Scudo.
+
+### Modifications to the Repositories and Vehicle Interactions
+
+To implement Scudo, one of Uptane's two repositories will need updating to
+enable it to store in-toto metadata alongside the images. The Image repository
+is recommended as its contents are typically signed for using offline keys,
```

Comment:
I'd just recommend stating this paragraph with recommending the Image repo instead of sounding like there is a choice first.

Reply:
Updated:

```
To implement Scudo, Uptane's Image repository will need to be updated to enable
it to store in-toto metadata alongside the images. While the Director repository
can also be used, the Image repository is recommended as its contents are
typically signed for using offline keys.
```